Kinvey-Xamarin/Offline/SQLiteCache.cs

```diff
@@ -502,50 +515,43 @@ public KinveyDeleteResponse Clear(Expression expr)
 		{
 			KinveyDeleteResponse kdr = new KinveyDeleteResponse();
 
+
 			try
 			{
 				int skipNumber = 0;
 				int takeNumber = 0;
 
 				var lambdaExpr = ConvertQueryExpressionToFunction(expr, ref skipNumber, ref takeNumber);
 
-				if (lambdaExpr != null)
+				if (lambdaExpr != null && skipNumber == 0)
 				{
-					try
-					{
-						List<T> matches = null;
+					List<T> results;
 
-						if (skipNumber > 0 && takeNumber > 0)
-						{
-							matches = dbConnectionSync.Table<T>().Where(lambdaExpr).Skip(skipNumber).Take(takeNumber).ToList();
-						}
-						else if (skipNumber > 0)
-						{
-							matches = dbConnectionSync.Table<T>().Where(lambdaExpr).Skip(skipNumber).ToList();
-						}
-						else if (takeNumber > 0)
-						{
-							matches = dbConnectionSync.Table<T>().Where(lambdaExpr).Take(takeNumber).ToList();
-						}
-						else
-						{
-							matches = dbConnectionSync.Table<T>().Where(lambdaExpr).ToList();
-						}
+					var query = dbConnectionSync.Table<T>();
+					if (lambdaExpr != null)
+					{
+						query = query.Where(lambdaExpr);
+					}
+					if (skipNumber != 0)
```

Comment:
@tejasranade This case will not be hit, because the outer conditional checks if `skipNumber` is 0.  Should this case be removed?

Also, do we have to handle the condition where there is no query, but there is a `Take` modifier?